### PR TITLE
Major changes in pgAdmin code to use ReactJS fully. Changes include:

### PR DESCRIPTION
**Pull request by adityatoshniwal**
Major changes in pgAdmin code to use ReactJS fully. Changes include:
1. Replace the current layout library wcDocker with ReactJS based rc-dock. https://github.com/pgadmin-org/pgadmin4/issues/6479
2. Have close buttons on individual panel tabs instead of common. https://github.com/pgadmin-org/pgadmin4/issues/2821
3. Changes in the context menu on panel tabs - Add close, close all and close others menu items. https://github.com/pgadmin-org/pgadmin4/issues/5394
4. Allow closing all the tabs including SQL and Properties. https://github.com/pgadmin-org/pgadmin4/issues/4733
5. Changes in docking behaviour of different tabs based on user requests and remove lock layout menu.
6. Fix an issue where scroll position of panels was not remembered on Firefox. https://github.com/pgadmin-org/pgadmin4/issues/2986
7. Reset layout now will not require page refresh and is done spontaneously.
8. Use zustand store for storing preferences instead of plain JS object. This will help reflecting preferences immediately.
9. The above fix incorrect format (no indent) of SQL stored functions/procedures. https://github.com/pgadmin-org/pgadmin4/issues/6720
10. New version check is moved to an async request now instead of app start to improve startup performance.
11. Remove jQuery and Bootstrap completely.
12. Replace jasmine and karma test runner with jest. Migrate all the JS test cases to jest. This will save time in writing and debugging JS tests.
13. Other important code improvements and cleanup.

### CODE REVIEW

File: web/regression/javascript/erd/ui_components/ERDTool.spec.js [hunk 1]
```diff
@@ -1,505 +1,472 @@
...
-    bodyInstance.onNoteClick();
-    expect(bodyInstance.setState).toHaveBeenCalledWith({
-      note_node: noteNode,
-      note_open: true,
-    });
-  });
+  it('dummy', ()=>{});
 });
+
+// import React from 'react';
```
Comment (khushboovashi) [inline web/regression/javascript/erd/ui_components/ERDTool.spec.js:19]: Remove comments.

File: web/webpack.config.js [hunk 2]
```diff
@@ -412,10 +408,10 @@ module.exports = [{
       exclude: [/node_modules/, /vendor/],
       use: {
         loader: 'babel-loader',
-        options: {
-          presets: [['@babel/preset-env', {'modules': 'commonjs', 'useBuiltIns': 'usage', 'corejs': 3}], '@babel/preset-react', '@babel/preset-typescript'],
-          plugins: ['@babel/plugin-proposal-class-properties', '@babel/proposal-object-rest-spread'],
-        },
+        // options: {
```
Comment (khushboovashi) [inline web/webpack.config.js:411]: Remove the code if not required

**Review by khushboovashi — CHANGES_REQUESTED** [review]
- Dashboard and System Stats graphs are broken.
- Close all the Panels, now open Properties and Dashboard panels. Refresh the page, Dashboard panel will be missing.
- If I re-arrange the panels and keep the properties tab in vertical right, the properties of the node doesn't display even if I select tree node again.

File: web/regression/javascript/erd/ui_components/ERDTool.spec.js (hunk 1 above)
Comment (adityatoshniwal) [reply web/regression/javascript/erd/ui_components/ERDTool.spec.js:19]: This particular test file is intentionally commented and I have mentioned the reason in the same file.

**Comment by adityatoshniwal** [comment]
> * Dashboard and System Stats graphs are broken.
> * Close all the Panels, now open Properties and Dashboard panels. Refresh the page, Dashboard panel will be missing.
> * If I re-arrange the panels and keep the properties tab in vertical right, the properties of the node doesn't display even if I select tree node again.

I have fixed the above issues.

**Review by khushboovashi — CHANGES_REQUESTED** [review]
- With all the open panels, open query tool, now move Properties tab (or any tab) to the bottom, query tool tab will be blank.
- Remove Lock Layout reference from menu_bar and preferences documentation.

**Comment by nikhil-mohite** [comment]
UI Issue found in testing:
1. The PSQL Tool icon should be hidden on the browser explore menu bar if `ENABLE_PSQL=False` in server mode, currently it is visible even value is set to False
image

Older version
image

**Pull request merged**